core/src/execution/datafusion/expressions/scalar_funcs.rs

```diff
@@ -366,7 +364,12 @@ fn spark_round(
                 let (precision, scale) = get_precision_scale(data_type);
                 make_decimal_array(array, precision, scale, &f)
             }
-            _ => make_scalar_function(math_expressions::round)(args),
+            DataType::Float32 | DataType::Float64 => {
+                Ok(ColumnarValue::Array(math_expressions::round(&[
+                    array.clone()
+                ])?))
+            }
+            dt => exec_err!("Not supported datatype[{dt}] for ROUND"),
```

Comment:
Hm, why wrap `dt` with `[]`? I don't see there is something in Comet or DataFusion.